### PR TITLE
fix typos

### PR DESCRIPTION
**Pull request by s1037989**
In addition to the fix typo requests, I have some questions that might help improve the documentation:

---

In the Synopsis, you state:

> The synopsis is split into three parts: The two first is for model developers, and the last is for the developer using the models.

There is a section header titled "Synopsis" so I would think that the 3 parts you're referring to are in this very small Synopsis, but I don't think that's the case.  If it's referring to 3 parts within the whole document, perhaps it could be reworded to "This documentation is split into three parts:" and if that is the case then I'm not sure where the three parts are.  Could you add markers to signify the three parts?

If you help me to understand the 3 sections, I'd be happy to apply the clarity through a PR.

---

For the table, you state:

> Used to define a table name. The default is to decamelize the last part of the class name and add "s" at the end, unless it already has "s" at the end.

Perhaps you could use Lingua::EN::Inflect to do this properly, for example with an example of App::Model::Box -> boxes instead of boxs?  Perhaps you could even do this only if Lingua::EN::Inflect is installed so as to not make that module a strict requirement?  However, that would be a bad idea because the code would expect boxes because the developer had the module installed but the user wouldn't have it installed and need boxs.  Perhaps setting a Mad::Mapper attribute of "proper_plural" to true would turn on this feature for the developer and then therefore require the module?

If you're interested in this I'd be willing to issue a pull request for this.  I think I could handle it completely on my own, but I'd be interested to know what you'd like to name the attribute.  Or if you'd have other ideas about how to go about this.

---

You refer often to "sst".  What does this mean?  Could you define it somewhere in the docs?  If you let me know, I'd be willing to find a place for it.

### CODE REVIEW

**Comment by s1037989** [comment]
Woops, it looks like an attribute won't work...  Can we do it with an import flag?  I'm not sure how that would look...

Current:
`use Mad::Mapper -base;`

Possible:
`use Mad::Mapper -base -proper_plural;`

Would that be an acceptable form?

**Comment by s1037989** [comment]
> `use Mad::Mapper -base -proper_plural;`

That isn't a *valid* form!  :(  Would have to be:

`use Mad::Mapper -base '-proper_plural';`

Or what about:

`use Mad::Mapper -base_pp;`

Just for shits and grins, I pulled together [something that works](https://github.com/s1037989/mad-mapper/commit/e7d42c0ee3a0518cad79f1364d59ec23e1e0255c), complete with documentation and tests!  :)

**Comment by s1037989** [comment]
Woops, I didn't mean to add that -proper_plural feature into this PR.  I don't know how that happened.  Can you accept the PR while rejecting that specific commit?

**Comment by jhthorsen** [comment]
I don't know if I answer all your questions. I think it would be easier if you ask me in #darkcode on IRC.

"The synopsis is split into three parts" - That is simply not true anymore. It used to have three sections, until i created the guides.

"Lingua::EN::Inflect" - Yeah... I couldn't find a case where I needed it, so I didn't want to add that as a dependency. I don't see when a table would be called "box", but I will use Lingua::EN::Inflect if you have a real example that is not covered. I don't care about "proper_plural". Let's just make it work properly.

"sst" - I though that was a standard database statement abbreviation, but I can't find it anywhere :disappointed: (I will ask)

**Comment by s1037989** [comment]
1. Great!  Let me know if you want help with that documentation clean up.
  1. I have a Model called Inventory and Inventorys wouldn't be correct.  :)
  1. Thanks!

**Comment by jhthorsen** [comment]
1. Yeah, any help is greatly appreciated! Just make something up :+1: 
2. Ah! Let's add Lingua::EN::Inflect. (I will add some comments inline)

File: lib/Mad/Mapper.pm [hunk 1]
```diff
@@ -253,16 +255,26 @@ sub import {
   }
 
   if ($flag) {
+    my @attr = @_;
     my $caller = caller;
     my $table = Mojo::Util::decamelize((split /::/, $caller)[-1]);
-    $table =~ s!s?$!s!;    # user => users
+    if ( grep { $_ eq '-proper_plural' } @attr ) {
```
Comment (jhthorsen) [inline lib/Mad/Mapper.pm]: Let's keep it simple: Just use Lingua::EN::Inflect in all cases. Also add Lingua::EN::Inflect to "cpanfile".

**Comment by jhthorsen** [comment]
I would also much appreciate if you could break this into two PR: One for the doc patch and one for the lingua change.

**Comment by s1037989** [comment]
Will do!  That was definitely an accident.  I didn't expect GitHub to automatically add new committed files into an already submitted PR.  Weird...

**Comment by jhthorsen** [comment]
@s1037989: yeah, it's very cool: you can rebase and force push to the PR :+1:

**Comment by s1037989** [comment]
I have some questions about rebase-ing if you don't mind.  I'm on <http://demo.convos.by/magnet/%23/convos>.  I really need to get myself my own convos server installed so I can more reliably have access to IRC...  :/

File: lib/Mad/Mapper.pm [hunk 2]
```diff
@@ -12,20 +12,20 @@ Mad::Mapper - Map Perl objects to PostgreSQL, MySQL or SQLite data
...
-  $ cpanm Mojo::Pg # Mad::Mapper now support postgres!
+  $ cpanm Mojo::Pg # Mad::Mapper now supports postgres!
 
 THIS MODULE IS EXPERIMENTAL. It is in use in production though, so
 big changes will not be made without extreme consideration.
 
 =head1 SYNOPSIS
 
-The synopsis is split into three parts: The two first is for model developers,
+The synopsis is split into three parts: The first two are for model developers,
```
Comment (jhthorsen) [inline lib/Mad/Mapper.pm]: I suggest that we just delete this line.

**Comment by jhthorsen** [comment]
Awesome! Thank you :+1:

**Pull request merged**